### DISCUSSION TITLE
既然 deposit 存币就是一个方法，那我可以给自己存100w个币吗

### DISCUSSION THREAD

**Discussion by hi-chenerge** [NonceGeek/Move-Camp#262] [Q&A] [answered]
写个合约，里面就一句话

`coin::deposit(address_of(account), Coin<AptosCoin> { value: 1000000 });`

不就是调用这个方法吗？谁写个程序就随便调用呗，

这里啥验证也都没有是不是

**Comment by qiwihui** [comment, marked as answer]
`deposit` 的作用是把币存入一个账户，前提是要有对应币的实体（比如价值 10 的 APT 币）。move 的机制保证结构体 `struct Coin<phantom CoinType> has store { value: u64} `只能在定义的包（coin包）中处理，所以你在自己写的包中调用 `Coin<AptosCoin> { value: 100000} ` 这种是会报错的。而要创建币，只能调用 coin::mint 方法，这个方法是需要传入 MintCapability 能力的，这个能力在初始化 Aptos 这个币的时候应该是被aptos这个系统拿着了，所以你没法自己mint Aptos这个币。 你可以看看这个mint的代码 https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/coin.move#L476
